### DISCUSSION TITLE
How do you approach a problem that you can't solve

### DISCUSSION THREAD

**Discussion by Koloru** [ouuan/ouuan#26] [Q&A (English)] [answered]
Hello I'm a self-taught junior developer and I was asking how do you approach problems in your codebase that you are unsure of solving.

currently at my job we have very big codebases(800 to 1500 lines) and I am always unsure of tackling them

Thank you.

**Comment by ouuan** [comment, marked as answer]
Your question is very general and I'm not sure how to answer it. But reading http://www.catb.org/~esr/faqs/smart-questions.html could be helpful. The "Before You Ask" section provides some general methods to solve the problems, and other sections help you ask better questions when you can't solve the problem on your own.
